Separate target and gateway

martinthomson opened the pull request:
This talks about the odd case where target and gateway are separate. The primary insight here is that the target is just a regular HTTP resource and the requests from the gateway aren't special.  Problems only occur if the gateway is privileged in some way.

If abuse comes via the gateway, then the target would ordinary block the client, which tends to affect a bunch of bystanders in this model.  So there is pressure to do something special for the gateway.  In that case, the target is trusting the gateway in some way.  Explain that.

The heading here is the only case where we use the short names of the entities, but I can't stand short headings.

martinthomson commented on draft-ietf-ohai-ohttp.md:
```diff
@@ -1134,6 +1134,24 @@ modified by a network attacker.  Note that a request could be forwarded without
+Gateway Resource is an ordinary HTTP resource.  A Target Resource can privilege
+requests that are forwarded by a given Oblivious Gateway Resource if it trusts
+the operator of the Oblivious Gateway Resource to only forward requests that
+meet the expectations of the Target Resource.
```
```suggestion
meet the expectations of the Target Resource.  Otherwise, the Target Resource
treats requests from an Oblivious Gateway Resource no differently than any
other HTTP client.
```

evyncke commented:
The changes are a big step forward in the right direction, but the flow seems a little weird as the text in the first part of the security section is about co-located gateway/target. **Strongly** suggest to move the added new text at the end of the whole section and add a forward reference in the text:

`In this section, a deployment where there are three entities is considered:` (from the intro in the security section)

evyncke commented:
And, just to be clear, the text addresses my concern. My comment about the flow is just editorial.

martinthomson commented:
I've taken the suggestion to move the text (with forward references) and made a few tweaks:

1. Rebased.
2. Moved and rewritten some text from the top of the security considerations that better fit the new section.
3. Added a forward reference to this section.
4. Moved and rewritten text added in #281 to the applicability section.  (Originally this was just to correct a misspelling, but I think this makes more sense in the new home.)

@chris-wood, can you double-check my work.  With this, I think we're ready to revise the draft and ask the IESG to finalize their ballot.

chris-wood commented on draft-ietf-ohai-ohttp.md:
```diff
@@ -1412,6 +1415,35 @@ content is processed.  Consequently, the security considerations of {{Section 17
+
+An Oblivious Gateway Resource could implement policies that improve the ability
+of the Target Resource to implement policy exemptions, such as only forwarding
+requests toward specific Target Resources.
```
```suggestion
requests toward specific Target Resources according to an allowlist; see {{server-responsibilities}}.
```
